File: scripts/capture_screenshots.mjs
/**
 * Capture GramSeva UI screenshots for executive presentation.
 * Requires: dev server on http://localhost:5173 (npm run dev)
 * Run: npx playwright install chromium && node scripts/capture_screenshots.mjs
 */
import { chromium } from "playwright";
import { mkdir, writeFile } from "fs/promises";
import path from "path";
import { fileURLToPath } from "url";

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const OUT = path.join(__dirname, "../docs/presentation-screenshots");
const BASE = process.env.GRAMSEVA_URL || "http://localhost:5173";

const DEMO_USER = {
  id: "USR-SCREENSHOT",
  name: "R. Lakshmi",
  role: "Sarpanch",
  phone: "9876543210",
  district: "Yadadri Bhuvanagiri",
  mandal: "Yadagirigutta",
  village: "Yadagirigutta Town",
  ward: "4",
  registeredAt: new Date().toISOString(),
};

async function seedLocalStorage(page) {
  await page.evaluate(
    ({ user, keys }) => {
      const users = [user];
      localStorage.setItem("tg_grama_seva_users", JSON.stringify(users));
      localStorage.setItem("tg_grama_seva_session", user.phone);
      localStorage.removeItem("tg_grama_seva_demo_seeded");
    },
    { user: DEMO_USER, keys: {} }
  );
}

async function clickSidebar(page, label) {
  const btn = page.locator("button", { hasText: label }).first();
  await btn.waitFor({ state: "visible", timeout: 15000 });
  await btn.click();
  await page.waitForTimeout(800);
}

async function capture(page, name, opts = {}) {
  const file = path.join(OUT, `${name}.png`);
  if (opts.selector) {
    const el = page.locator(opts.selector).first();
    await el.waitFor({ state: "visible", timeout: 15000 });
    await el.screenshot({ path: file });
  } else if (opts.clip) {
    await page.screenshot({ path: file, clip: opts.clip });
  } else {
    await page.screenshot({ path: file, fullPage: false });
  }
  console.log("  ✓", name);
  return file;
}

async function main() {
  await mkdir(OUT, { recursive: true });
  const browser = await chromium.launch();
  const context = await browser.newContext({ deviceScaleFactor: 2 });

  // Mobile — citizen experience
  const mobile = await context.newPage();
  await mobile.setViewportSize({ width: 390, height: 844 });
  await mobile.goto(`${BASE}/app`, { waitUntil: "networkidle", timeout: 60000 });
  await seedLocalStorage(mobile);
  await mobile.reload({ waitUntil: "networkidle" });
  await mobile.waitForTimeout(1500);

  await capture(mobile, "01-home-mobile", { selector: "main" });
  await clickSidebar(mobile, "Report Issue");
  await capture(mobile, "02-report-mobile", { selector: "main" });
  await clickSidebar(mobile, "Village Logs");
  await capture(mobile, "03-issues-mobile", { selector: "main" });
  await clickSidebar(mobile, "Emergency & Help");
  await capture(mobile, "04-emergency-mobile", { selector: "main" });
  await clickSidebar(mobile, "AI Assistant");
  await capture(mobile, "05-vikas-mobile", { selector: "main" });

  // Desktop — official / analytics
  const desktop = await context.newPage();
  await desktop.setViewportSize({ width: 1440, height: 900 });
  await desktop.goto(`${BASE}/app`, { waitUntil: "networkidle", timeout: 60000 });
  await seedLocalStorage(desktop);
  await desktop.reload({ waitUntil: "networkidle" });
  await desktop.waitForTimeout(1500);

  await capture(desktop, "06-home-desktop", { selector: "main" });
  await clickSidebar(desktop, "Official Dashboard");
  await capture(desktop, "07-official-dashboard", { selector: "main" });
  await clickSidebar(desktop, "Village Analytics");
  await capture(desktop, "08-analytics", { selector: "main" });
  await clickSidebar(desktop, "Welfare Hub");
  await capture(desktop, "09-welfare", { selector: "main" });
  await clickSidebar(desktop, "News & Announcements");
  await capture(desktop, "10-news", { selector: "main" });

  // Login screen (logged out)
  const login = await context.newPage();
  await login.setViewportSize({ width: 1440, height: 900 });
  await login.goto(`${BASE}/app`, { waitUntil: "networkidle" });
  await login.evaluate(() => localStorage.clear());
  await login.reload({ waitUntil: "networkidle" });
  await login.waitForTimeout(1200);
  await capture(login, "11-login", { selector: "body" });

  // Standalone emergency page
  const emerg = await context.newPage();
  await emerg.setViewportSize({ width: 390, height: 844 });
  await emerg.goto(`${BASE}/emergency`, { waitUntil: "networkidle" });
  await emerg.waitForTimeout(1200);
  await capture(emerg, "12-emergency-standalone", { selector: "body" });

  await browser.close();
  console.log("\nScreenshots saved to:", OUT);
}

main().catch((err) => {
  console.error("Screenshot capture failed:", err.message);
  console.error("Ensure dev server is running: npm run dev");
  process.exit(1);
});

File: scripts/capture_screenshots.py
#!/usr/bin/env python3
"""Capture GramSeva UI screenshots for executive presentation."""

import subprocess
import sys
from pathlib import Path

try:
    from playwright.sync_api import sync_playwright
except ImportError:
    subprocess.check_call([sys.executable, "-m", "pip", "install", "playwright", "-q"])
    subprocess.check_call([sys.executable, "-m", "playwright", "install", "chromium"])
    from playwright.sync_api import sync_playwright

ROOT = Path(__file__).resolve().parent.parent
OUT = ROOT / "docs" / "presentation-screenshots"
BASE = "http://localhost:5173"

DEMO_USER = {
    "id": "USR-SCREENSHOT",
    "name": "R. Lakshmi",
    "role": "Sarpanch",
    "phone": "9876543210",
    "district": "Yadadri Bhuvanagiri",
    "mandal": "Yadagirigutta",
    "village": "Yadagirigutta Town",
    "ward": "4",
    "registeredAt": "2026-01-01T00:00:00.000Z",
}

VIEWS = [
    ("home", "01-home-mobile", "06-home-desktop"),
    ("report", "02-report-mobile", None),
    ("issues", "03-issues-mobile", None),
    ("emergency", "04-emergency-mobile", None),
    ("vikas", "05-vikas-mobile", None),
    ("official", None, "07-official-dashboard"),
    ("analytics", None, "08-analytics"),
    ("welfare", None, "09-welfare"),
    ("news", None, "10-news"),
]

SIDEBAR = {
    "home": "Home",
    "report": "Report Issue",
    "issues": "Village Logs",
    "emergency": "Emergency & Help",
    "vikas": "AI Assistant",
    "official": "Official Dashboard",
    "analytics": "Village Analytics",
    "welfare": "Welfare Hub",
    "news": "News & Announcements",
}


def seed(page):
    page.evaluate(
        """(user) => {
            localStorage.setItem('tg_grama_seva_users', JSON.stringify([user]));
            localStorage.setItem('tg_grama_seva_session', user.phone);
            localStorage.removeItem('tg_grama_seva_demo_seeded');
        }""",
        DEMO_USER,
    )


def nav_to(page, key: str):
    page.get_by_role("button", name=SIDEBAR[key]).first.click(timeout=15000)
    page.wait_for_timeout(1000)


def shot(page, name: str, selector="main"):
    path = OUT / f"{name}.png"
    page.locator(selector).first.screenshot(path=str(path))
    print(f"  ✓ {name}")


def main():
    OUT.mkdir(parents=True, exist_ok=True)
    with sync_playwright() as p:
        browser = p.chromium.launch()
        page = browser.new_page()
        page.set_viewport_size({"width": 1440, "height": 900})
        page.goto(f"{BASE}/app", wait_until="networkidle", timeout=60000)
        seed(page)
        page.reload(wait_until="networkidle")
        page.wait_for_timeout(1500)

        for key, mobile_name, desktop_name in VIEWS:
            nav_to(page, key)
            if desktop_name:
                shot(page, desktop_name)
            if mobile_name:
                page.set_viewport_size({"width": 390, "height": 844})
                page.wait_for_timeout(500)
                shot(page, mobile_name)
                page.set_viewport_size({"width": 1440, "height": 900})
                page.wait_for_timeout(300)

        login = browser.new_page()
        login.set_viewport_size({"width": 1440, "height": 900})
        login.goto(f"{BASE}/app", wait_until="networkidle")
        login.evaluate("localStorage.clear()")
        login.reload(wait_until="networkidle")
        login.wait_for_timeout(1200)
        shot(login, "11-login", "body")

        emerg = browser.new_page()
        emerg.set_viewport_size({"width": 390, "height": 844})
        emerg.goto(f"{BASE}/emergency", wait_until="networkidle")
        emerg.wait_for_timeout(1200)
        shot(emerg, "12-emergency-standalone", "body")

        browser.close()
    print(f"\nScreenshots saved to: {OUT}")


if __name__ == "__main__":
    main()
